netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -119,11 +119,12 @@ public void ping(PingCallback callback, Executor executor) {
   }
 
   @Override
-  public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata.Headers headers,
+  public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers,
       ClientStreamListener listener) {
     Preconditions.checkNotNull(method, "method");
     Preconditions.checkNotNull(headers, "headers");
     Preconditions.checkNotNull(listener, "listener");
+ // FIXME: Remove authority
```

Comment:
Remove the authority key here, and if it was non-null, pass it into `convertClientHeaders()` instead of the class-field `authority`.